src/vanilla/utils/selectAtom.ts

```diff
@@ -29,6 +29,11 @@ export function selectAtom<Value, Slice>(
   return memo3(
     () => {
       const EMPTY = Symbol()
+      const refAtom = atom(() => ({
```

Comment:
We used to use `refAtom` to implement `selectAtom` and it's improved since then. We should be able to implement without `refAtom`. Otherwise, it comes with more hacks.
BTW, using `promise.status` might be a better hack.

Reply:
there is potentially a weakmap solution. I need to think more on this.

Reply:
I would probably prefer the weakmap solution.